From 67a93f418f309209d5999f0c85b20eb5f0e987f2 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 23 Nov 2017 01:31:32 +0000
Subject: [PATCH] storage: No, nope, and even more nope.

Perfect example of upstreams doing crazy stuff they shouldn't be.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 storage/tokudb/CMakeLists.txt           |  2 --
 storage/tokudb/PerconaFT/CMakeLists.txt | 25 -------------------------
 2 files changed, 27 deletions(-)

diff --git a/storage/tokudb/CMakeLists.txt b/storage/tokudb/CMakeLists.txt
index c959bf00..077960b0 100644
--- a/storage/tokudb/CMakeLists.txt
+++ b/storage/tokudb/CMakeLists.txt
@@ -49,8 +49,6 @@ MARK_AS_ADVANCED(TOKUDB_DATA)
 MARK_AS_ADVANCED(TOKU_DEBUG_PARANOID)
 MARK_AS_ADVANCED(USE_VALGRIND)
 MARK_AS_ADVANCED(XZ_SOURCE_DIR)
-MARK_AS_ADVANCED(gcc_ar)
-MARK_AS_ADVANCED(gcc_ranlib)
 ############################################
 
 # pick language dialect
diff --git a/storage/tokudb/PerconaFT/CMakeLists.txt b/storage/tokudb/PerconaFT/CMakeLists.txt
index d42525f1..b520dd68 100644
--- a/storage/tokudb/PerconaFT/CMakeLists.txt
+++ b/storage/tokudb/PerconaFT/CMakeLists.txt
@@ -19,31 +19,6 @@ if (DEFINED MYSQL_PROJECT_NAME_DOCSTRING)
   endif ()
 endif ()
 
-## Versions of gcc >= 4.9.0 require special version of 'ar' and 'ranlib' for
-## link-time optimizations to work properly.
-##
-## From https://gcc.gnu.org/gcc-4.9/changes.html:
-##
-##     When using a linker plugin, compiling with the -flto option now
-##     generates slim objects files (.o) which only contain intermediate
-##     language representation for LTO. Use -ffat-lto-objects to create
-##     files which contain additionally the object code. To generate
-##     static libraries suitable for LTO processing, use gcc-ar and
-##     gcc-ranlib; to list symbols from a slim object file use
-##     gcc-nm. (Requires that ar, ranlib and nm have been compiled with
-##     plugin support.)
-if ((CMAKE_CXX_COMPILER_ID STREQUAL GNU) AND
-    NOT (CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9.0"))
-  find_program(gcc_ar "gcc-ar")
-  if (gcc_ar)
-    set(CMAKE_AR "${gcc_ar}")
-  endif ()
-  find_program(gcc_ranlib "gcc-ranlib")
-  if (gcc_ranlib)
-    set(CMAKE_RANLIB "${gcc_ranlib}")
-  endif ()
-endif()
-
 include(TokuFeatureDetection)
 include(TokuSetupCompiler)
 include(TokuSetupCTest)
-- 
2.15.0

